### ISSUE TITLE
Specs

### ISSUE THREAD

**Issue by daraosn** [TxHash/ifttt-bridge#1]
Objective: Connect TxHash API to IFTTT, to be able to create recipes that react on payment notifications.

Specification:
Use the trigger (New Ethereum payment received) from https://ifttt.com/txhash and connect it to our TxHash API to track payments directly from IFTTT and react using a recipe.
i.e. When an Ethereum payment is received at address 0x123456789abcedf...., log it by adding it to a Google Spreedsheet (https://ifttt.com/google_sheets).

To do list:
- [ ] Setup a local IFTTT sample application https://platform.ifttt.com/docs
- [ ] Configure end points https://platform.ifttt.com/services/txhash/api/endpoint_tests
- [ ]  Connect API end points (IFTTT <> TXHASH)
  - [ ] Check to understand how to use the API https://github.com/TxHash/python-api-sample
  - [ ] When deploying a new IFTTT recipe, this ifttt-bridge should create a new tracker on TxHash with the address provided and a TxHash tracker action pointing back to IFTTT recipe action so it can react accordingly
  - [ ] Add simple tests if possible (optional for now, we can work together on this)

**Comment by daraosn** [comment]
@ernaneluis Let's use [ExpressJS](https://github.com/expressjs) or [Restify](https://github.com/restify/node-restify) (recommended as it's simpler to use) plus [RequestJS]
(https://github.com/request/request) for the stack

**Comment by daraosn** [comment]
Also, this might work still, not sure (it's 3 years old) https://www.npmjs.com/package/ifttt

**Comment by ernaneluis** [comment]
I think we should use [AXIOS](https://www.npmjs.com/package/axios) or request-promise
since the  request do not support Promises.

**Comment by ernaneluis** [comment]
I need access for the https://ifttt.com/txhash

**Comment by ernaneluis** [comment]
Where is the doc of TXHASH API?

**Comment by ernaneluis** [comment]
Why we need to do on python? This python code just do a HTTP request. nothing fancy

**Comment by daraosn** [comment]
@ernaneluis 

1. Yes, you can use `request-promise` (the original prototype I started to build with this package).
2. Check now https://ifttt.com/txhash 
3. Python code is just an example so you know how the API is working as it's not yet documented. Those are the main calls you need to build this.

**Comment by ernaneluis** [comment]
ok so I can replicate the pyton code into node js + express.js right?

**Comment by daraosn** [comment]
@ernaneluis yes, feel free to use the live version of http://txhash.com as it's on alpha anyway and the database will be cleaned once we release the beta

**Comment by daraosn** [comment]
also if you need to test and send ETH, you can use small amounts so you don't consume your ETH... any amount, even as small as 0.000000001 ETH or less will work, and pay 0.1 Gwei... cost should be less than a cent... I will work on activating the test net soon (and a staging server)
